client/components/CustomVocabulariesFields.tsx

```diff
@@ -32,7 +33,10 @@ export default class CustomVocabulariesFields extends React.PureComponent<IProps
             errors,
             diff,
         } = fieldProps;
-        const language = get(diff, 'language') || getUserInterfaceLanguageFromCV();
+
+        const language = MAIN_LANGUAGE ?? get(diff, 'language') ?? getUserInterfaceLanguageFromCV();
```

Comment:
Hi @MarkLark86 I want to consult regarding the approach to fix this ticket. In multilingual mode there is the [component](https://github.com/superdesk/superdesk-planning/assets/2076953/17a9ab04-43b8-4803-99ae-2578d5021a34) to select the main language. When the language is changed, it saves it to the state. BUT when we are rendering a vocabulary select, we use `item.language`. Should we fix it by always updating `item.language` when the value of language select component changes? Or should we rather not touch `item.language` and use that state variable that stores the main language?

Reply:
Hi @tomaskikutis & @dzonidoo 

This `Main Language` selector is purely for view purposes and nothing to do with the data layer, therefor my recommendation would be to use the Editor's state.

This component is rendered by the Editor, using the [EditorFieldCustomVocabularies](https://github.com/superdesk/superdesk-planning/blob/develop/client/components/fields/editor/CustomVocabularies.tsx) component. The Editor passes in the `language` that should be used for rendering (see [IEditorFieldProps](https://github.com/superdesk/superdesk-planning/blob/develop/client/interfaces.ts#L1494) for the interface).

My suggested solution here would be to:
* Remove any language logic from within this component
* Add a `language` prop to this component
* Pass through the `language` prop in the `EditorFieldCustomVocabularies` component

This way the language used in this component is controlled elsewhere, in this case the Editor itself.